rocketpy/simulation/flight.py

```diff
@@ -662,17 +665,20 @@ def __init__(
             # Initialize phase time nodes
             phase.TimeNodes = TimeNodes()
             # Add first time node to permanent list
-            phase.TimeNodes.add_node(phase.t, [], [])
+            phase.TimeNodes.add_node(phase.t, [], [], [])
             # Add non-overshootable parachute time nodes
             if self.time_overshoot is False:
                 phase.TimeNodes.add_parachutes(
                     self.parachutes, phase.t, phase.time_bound
                 )
+                phase.TimeNodes.add_sensors(
+                    self.rocket.sensors, phase.t, phase.time_bound
+                )
                 phase.TimeNodes.add_controllers(
                     self._controllers, phase.t, phase.time_bound
                 )
```

Comment:
Could we merge these 3 functions into a single one? They all receive the same inputs...

```
                phase.TimeNodes.add_parachutes(
                    self.parachutes, phase.t, phase.time_bound
                )
                phase.TimeNodes.add_sensors(
                    self.rocket.sensors, phase.t, phase.time_bound
                )
                phase.TimeNodes.add_controllers(
                    self._controllers, phase.t, phase.time_bound
                )
```

Reply:
I think we gotta make a separate PR refactoring TimeNodes and FlightPhases to do this. It would take a lot of work to guarantee any changes in this classes do not break some specific case in the flight class